homeassistant/components/evohome.py

```diff
@@ -0,0 +1,606 @@
+                "Check your username (%s), and password are correct."
+                "Unable to continue. Resolve any errors and restart HA.",
+                domain_data['params'][CONF_USERNAME]
+            )
+        else:
+            # Otherwise, it may be enough to back off and try again later.
+            raise PlatformNotReady(err)
+
+    finally:  # Redact username, password as no longer needed.
+        domain_data['params'][CONF_USERNAME] = 'REDACTED'
+        domain_data['params'][CONF_PASSWORD] = 'REDACTED'
+
+    domain_data['client'] = client
+
+    # Redact any installation data we'll never need.
+    if client.installation_info[0]['locationInfo']['locationId'] != 'REDACTED':
+        for loc in client.installation_info:
+            loc['locationInfo']['locationId'] = 'REDACTED'
+            loc['locationInfo']['streetAddress'] = 'REDACTED'
+            loc['locationInfo']['city'] = 'REDACTED'
+            loc['locationInfo']['locationOwner'] = 'REDACTED'
+            loc[GWS][0]['gatewayInfo'] = 'REDACTED'
+
+    # Pull down the installation configuration...
+    loc_idx = domain_data['params'][CONF_LOCATION_IDX]
+
+    try:
+        domain_data['config'] = client.installation_info[loc_idx]
+
+    # IndexError: configured location index is outside the range
+    except IndexError:
+        _LOGGER.warning(
+            "setup(): Config parameter, '%s'= %s , is out of range (0-%s), "
+            "continuing with '%s' = 0.",
+            CONF_LOCATION_IDX,
+            loc_idx,
+            len(client.installation_info) - 1,
+            CONF_LOCATION_IDX
+        )
+
+        domain_data['params'][CONF_LOCATION_IDX] = 0
+        domain_data['config'] = client.installation_info[0]
+
+    domain_data['status'] = {}
+
+    if _LOGGER.isEnabledFor(logging.DEBUG):
+        _LOGGER.debug(
+            "setup(): Location/TCS (temp. control system) used is: %s [%s]",
+            domain_data['config'][GWS][0][TCS][0]['systemId'],
+            domain_data['config']['locationInfo']['name']
+        )
+        # Some platform data needs further redaction before being logged.
+        tmp = dict(domain_data['config'])
+        tmp['locationInfo']['postcode'] = 'REDACTED'
+
+        _LOGGER.debug("setup(): domain_data['config']: %s", tmp)
+
+    # We have the platofrom configuration, but no state as yet, so...
+    def _first_update(event):
+        """Let the controller know it can obtain it's first update."""
+    # Send a message to the hub to do its first update()
+        pkt = {
+            'sender': 'setup()',
+            'signal': 'update',
+            'to': EVO_MASTER
+        }
+        hass.helpers.dispatcher.dispatcher_send(
+            DISPATCHER_EVOHOME,
+            pkt
+        )
+
+    hass.bus.listen(EVENT_HOMEASSISTANT_START, _first_update)
+
+    load_platform(hass, 'climate', DOMAIN)
+
+    return True
+
+
+class EvoEntity(ClimateDevice):
```

Comment:
Since this one is only supposed to be used inside the evohome climate platform, we should move it to the climate platform.

Reply:
OK, but I am going to submit subsequent PRs that will split them out again!

Reply:
was moved.